simulation/src/main/java/com/palantir/dialogue/core/TaggedMetrics.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.codahale.metrics.Clock;
+import com.codahale.metrics.Counter;
+import com.codahale.metrics.Meter;
+import com.codahale.metrics.SlidingTimeWindowArrayReservoir;
+import com.codahale.metrics.Timer;
+import com.palantir.tritium.metrics.registry.AbstractTaggedMetricRegistry;
+import com.palantir.tritium.metrics.registry.MetricName;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+public final class TaggedMetrics extends AbstractTaggedMetricRegistry {
```

Comment:
@carterkozak here's that FLUP from the earlier PR - turned out much easier than I was expecting!

Reply:
Ideally I'd prefer to use metrics from our production code, otherwise I'm worried we won't have the data to understand real use. We should be able to update existing timers like this in order to work properly with simulation: #359